lua/state_lua.cpp

```diff
@@ -256,7 +256,7 @@ void updateUserValueAndPush(
 
   // Still up to date?
   lua_getfield(L, -1, "__numUpdates");
-  if (lua_tointeger(L, -1) == s->numUpdates) {
+  if ((long) lua_tointeger(L, -1) == (long) s->numUpdates) {
```

Comment:
Maybe lua_tolong here

Reply:
lua_tolong seems to be unavailable. Will use lua_tonumber https://github.com/quakenet/newserv/blob/master/lua/lua.h#L91

Reply:
Ah I didn't realize. tonumber appears to give a double, so they original might be better. Sorry about that!

Reply:
#define lua_toint(a, b) (int) lua_tonumber(a, b)

so (long) lua_tonumber(a, b) is still better than what I originally had :)

Reply:
Ahaha wtf, feel free to merge then :)

Reply:
BTW if you want a totally sweet probably-not-functional Lua wrapper for BWAPI 2.3 -- I wrote https://code.google.com/archive/p/bwapi-lua/ back in like 2009 :D (and proceeded to never use it)